src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -20,239 +23,326 @@ public class ImmutableArrayTest : SimpleElementImmutablesTestBase
         private static readonly ImmutableArray<GenericParameterHelper> s_oneElementRefType = ImmutableArray.Create(new GenericParameterHelper(1));
         private static readonly ImmutableArray<string> s_twoElementRefTypeWithNull = ImmutableArray.Create("1", null);
 
-        [Fact]
-        public void Clear()
+        public static IEnumerable<object[]> Int32EnumerableData()
         {
-            Assert.Equal(ImmutableArray<int>.Empty, ImmutableArray.Create<int>().Clear());
-            Assert.Equal(ImmutableArray<int>.Empty, ImmutableArray.Create<int>(1).Clear());
-            Assert.Equal(ImmutableArray<int>.Empty, ImmutableArray.Create<int>(1, 2, 3).Clear());
+            // TODO: Standardize over Create vs new[].
+            yield return new object[] { new int[0] };
+            yield return new object[] { new[] { 1 } };
+            yield return new object[] { Enumerable.Range(1, 3) };
+            yield return new object[] { Enumerable.Range(4, 4) };
+            yield return new object[] { ImmutableArray.Create(2, 3, 5) };
         }
 
-        [Fact]
-        public void CreateEmpty()
+        public static IEnumerable<object[]> StrongBoxedInt32EnumerableData()
         {
-            Assert.Equal(ImmutableArray<int>.Empty, ImmutableArray.Create<int>());
-            Assert.Equal(ImmutableArray<int>.Empty, ImmutableArray.Create<int>(new int[0]));
+            // Once https://github.com/xunit/assert.xunit/pull/5 comes into corefx, all the StrongBox stuff can be removed.
+
+            return Int32EnumerableData()
+                .Select(array => array[0])
+                .Cast<IEnumerable<int>>()
+                .Select(enumerable => new object[]
+                {
+                    new StrongBox<IEnumerable<int>>(enumerable)
+                });
         }
 
-        [Fact]
-        public void CreateFromEnumerable()
+        public static IEnumerable<object[]> StrongBoxedSpecialInt32ImmutableArrayData()
         {
-            Assert.Throws<ArgumentNullException>("items", () => ImmutableArray.CreateRange((IEnumerable<int>)null));
+            // Once https://github.com/xunit/assert.xunit/pull/5 comes into corefx, all the StrongBox stuff can be removed.
```

Comment:
xUnit tries to format the collection for each test case by calling `GetEnumerator` and stringifying each of the values. For default immutable arrays, this is problematic because that methods throws an InvalidOperationException. This was fixed in https://github.com/xunit/assert.xunit/pull/5, but it looks like corefx hasn't picked up that change yet. I've worked around this issue temporarily by wrapping each enumerable in a `StrongBox<T>` so it won't get formatted.